pkg/apis/core/types_shoot.go

```diff
@@ -734,6 +734,17 @@ type Worker struct {
 	// Zones is a list of availability zones that are used to evenly distribute this worker pool. Optional
 	// as not every provider may support availability zones.
 	Zones []string
+	// Settings contains configurations for different worker-pools. Eg. MachineDrainTimeout, MachineHealthTimeout.
+	Settings *Settings
+}
+
+// Settings contains configurations for different worker-pools.
```

Comment:
Please refine

Reply:
done - https://github.com/gardener/gardener/commit/9aacdeb0348a4d06173bb54d71af652807da0846#diff-9a667891729b65e0ebbf3feef920fcd9R761 . Please suggest, if anything specific need to be added.